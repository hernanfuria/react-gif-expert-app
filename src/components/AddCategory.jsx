import { useState } from "react"
import PropTypes from 'prop-types';

export const AddCategory = ({handleAddCategory}) => {
    const [inputValue, setInputValue] = useState('')

    const handleInputChange = (event) => {
        setInputValue(event.target.value)
    }

    const handleSubmit = (event) => {
        event.preventDefault();
        if (inputValue != '') {
            handleAddCategory(inputValue);
            setInputValue('');
        }
    }

    return (
        <form onSubmit={(event) => handleSubmit(event)} aria-label="form">
            <input 
                type="text" 
                placeholder="Buscar GIFs" 
                value={inputValue}
                onChange={(event) => handleInputChange(event)}
            />
        </form>
  )
}

AddCategory.propTypes = {
    handleAddCategory: PropTypes.func.isRequired,
}