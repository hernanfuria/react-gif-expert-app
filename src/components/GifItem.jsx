import PropTypes from 'prop-types';

export const GifItem = ({id, title, url}) => {
    return (
        <div className='card'>
            <img src={url} alt={title} />
            <p>{title}</p>
        </div>
    )
}

GifItem.propTypes = {  // defines input type hints
    id: PropTypes.string.isRequired,
    title: PropTypes.string.isRequired,
    url: PropTypes.string.isRequired,
}
