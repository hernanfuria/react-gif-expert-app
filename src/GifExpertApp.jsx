import { useState } from 'react'
import { AddCategory, GifGrid } from './components';

export const GifExpertApp = () => {
    const [categories, setCategories] = useState([])

    const handleAddCategory = (cat) => {
        setCategories([cat, ...categories]);
    }

    const handleRemoveCategory = (cat) => {
        const newCategories = []
        for (const c of categories) {
            (c != cat) && newCategories.push(c);
        }

        setCategories(newCategories);
    }

    return (
        <>
            <h1>GifExpertApp</h1>

            <AddCategory 
                handleAddCategory={handleAddCategory} 
            />

            {/*<button aria-label='agregar' onClick={handleAddCategory}>Agregar</button>*/}
            {
                categories.map(category => {
                    return (<GifGrid 
                                key={category} 
                                category={category} 
                                handleRemoveCategory={handleRemoveCategory}
                            />)  
                })
            }
        </>
    )
}
