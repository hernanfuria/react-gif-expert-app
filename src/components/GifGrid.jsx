import PropTypes from 'prop-types';
import { GifItem } from './GifItem';
import { useFetchGifs } from '../hooks/useFetchGifs';


export const GifGrid = ({category, handleRemoveCategory}) => {
    const {images, isloading} = useFetchGifs(category);

    const handleDeleteButtonPressed = () => {
        console.log(`Borrando ${category}`)
        handleRemoveCategory(category);
    }
   
    return (
        <>
            <div className="category-header">
                <h3>{category}</h3>
                <button className='button-remove-category' onClick={handleDeleteButtonPressed}>Borrar</button>
            </div>
            {isloading && (<h2>Cargando...</h2>)}
            <div className='card-grid'>
                {images.map((image) => (
                    <GifItem 
                        key={image.id} 
                        {...image} 
                    />
                ))}
            </div>
        </>
    )
}

GifGrid.propTypes = {  // defines input type hints
    category: PropTypes.string.isRequired,
}

