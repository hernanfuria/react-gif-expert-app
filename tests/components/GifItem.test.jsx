import { render, screen } from "@testing-library/react"
import { GifItem } from "../../src/components"

describe('pruebas en GifItem', () => { 
    const id = 1
    const title = "test title" 
    const url = "https://media0.giphy.com/media/Yg12tqyJwylsk/giphy.gif?cid=4d294dfd4ewh1wkyi28ltzalazrn2woimqppy4d4l2ateyde&ep=v1_gifs_search&rid=giphy.gif&ct=g"

    const component = <GifItem id={id} title={title} url={url} /> 
    
    test('Debe matchear el snapshot', () => { 
        const {container} = render(component);
        expect(container).toMatchSnapshot();
     })

     test('Debe usar SRC y ALT correctos', () => { 
        render(component);
        // expect(screen.getByRole('img').src).toBe(url)
        const {src, alt} = screen.getByRole('img')
        expect(src).toBe(url)
        expect(alt).toBe(title)
      })

      test('Debe mostrar el titulo en el componente', () => { 
        render(component);
        expect(screen.getByText(title)).toBeTruthy();
       })
 })
