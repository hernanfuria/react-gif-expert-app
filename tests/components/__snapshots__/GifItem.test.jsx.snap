// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`pruebas en GifItem Debe matchear el snapshot 1`] = `
<div>
  <div
    class="card"
  >
    <img
      alt="test title"
      src="https://media0.giphy.com/media/Yg12tqyJwylsk/giphy.gif?cid=4d294dfd4ewh1wkyi28ltzalazrn2woimqppy4d4l2ateyde&ep=v1_gifs_search&rid=giphy.gif&ct=g"
    />
    <p>
      test title
    </p>
  </div>
</div>
`;
